tests/test_deprecated.py

```diff
@@ -75,24 +74,55 @@ class Pokemon(BaseModel):
         jp_name: str
 
     class PokemonList(BaseModel):
-        __root__: List[Pokemon]
+        root: List[Pokemon]
+
+        @root_validator(pre=True)
+        @classmethod
+        def populate_root(cls, values):
+            return {'root': values}
+
+        @model_serializer
+        def _serialize(self, info):
+            if info.mode == 'json':
+                return self.root
+            return self.__dict__
+
+        @classmethod
+        def model_modify_json_schema(cls, json_schema):
+            return json_schema['properties']['root']
```

Comment:
They do, but they'll have to rename the attribute anyway as it will raise an error when they try to create the model if it has a field called `__root__`.

One thing we could do to ease the transition would be to actually allow that as a field name (the only allowed dunder field name) and emit a DeprecationWarning. (We could add the appropriate model_serializer, model_modify_json_schema, and root validator). However, it seems to me we probably shouldn't do this? (I'm open to being convinced though.)

Considering this is just a "reference" migration, I don't know that it makes sense to add `__root__` as a property since, as I said, they'd need to rename the field anyway; users can do this if for some reason they are okay changing the field name but don't want to change the access pattern. (In principle, in that case, they may also want/need to add a setter for that property too.)

Reply:
okay, let's leave it as-is for now.